### PR TITLE
Add snapcraft-parser integration test.

### PR DESCRIPTION
**Pull request by josepht**
Add a "Happy Path" integration test for snapcraft-parser.
LP: #1590268

### CODE REVIEW

**Comment by snappy-m-o** [comment]
Can one of the admins verify this patch?

**Comment by snappy-m-o** [comment]
Can one of the admins verify this patch?

File: integration_tests/test_parser.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import os
+import subprocess
+
+from snapcraft.tests import TestCase
+
+
+class TestParser(TestCase):
+    '''Test bin/snapcraft-parser'''
```
Comment (elopio) [inline integration_tests/test_parser.py]: For consistency, the comments are with three double quotes.

File: integration_tests/test_parser.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import os
+import subprocess
+
+from snapcraft.tests import TestCase
+
+
+class TestParser(TestCase):
```
Comment (elopio) [inline integration_tests/test_parser.py]: This should inherit from integration_tests.TestCase

File: integration_tests/test_parser.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    '''Test bin/snapcraft-parser'''
+
+    def test_parser_basic(self):
+        '''Test snapcraft-parser basic usage'''
+        src_path = os.path.join(os.path.dirname(os.path.abspath(__file__)),
+                                '..', 'bin', 'snapcraft-parser')
```
Comment (elopio) [inline integration_tests/test_parser.py]: We need this test also to run from the installed path in autopkgtests. Take a look at integration_tests/__init__.py and add a self.snapcraft_parser_command.

File: integration_tests/test_parser.py [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+        args = ['python3', src_path, '--index',
```
Comment (elopio) [inline integration_tests/test_parser.py]: No need for python3 there.

**Comment by elopio** [comment]
Great, thanks.
I think I would like to get this to check parts of the output, but I don't like the current output. And I think I would like to check parts of the resulting file. Both things would fail if the index changes, so maybe it would be better to add a test later with a local index, and leave the one accessing the wiki simple. :+1: after the comments are fixed.

File: integration_tests/test_parser.py [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+                'https://wiki.ubuntu.com/snapcraft/parts?action=raw',
+                '--output', 'parts.yaml']
+        with open(os.path.devnull, 'w') as devnull:
+            subprocess.check_call(args, stderr=devnull, stdout=devnull)
```
Comment (elopio) [inline integration_tests/test_parser.py]: use subprocess.DEVNULL

File: integration_tests/test_parser.py (hunk 1 above)
Comment (josepht) [inline integration_tests/test_parser.py]: Fixed.

File: integration_tests/test_parser.py (hunk 2 above)
Comment (josepht) [inline integration_tests/test_parser.py]: Fixed.

File: integration_tests/test_parser.py (hunk 3 above)
Comment (josepht) [inline integration_tests/test_parser.py]: Fixed.

File: integration_tests/test_parser.py (hunk 4 above)
Comment (josepht) [inline integration_tests/test_parser.py]: Removed

**Comment by josepht** [comment]
@elopio I've updated based on your comments.  Thanks for the review.

**Comment by elopio** [comment]
:+1:

**Comment by snappy-m-o** [comment]
Can one of the admins verify this patch?

**Comment by snappy-m-o** [comment]
Can one of the admins verify this patch?

**Comment by sergiusens** [comment]
OK to test

**Pull request merged**